homeassistant/components/renault/services.py

```diff
@@ -0,0 +1,194 @@
+"""Support for Renault services."""
+from __future__ import annotations
+
+from datetime import datetime
+import logging
+from types import MappingProxyType
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+LOGGER = logging.getLogger(__name__)
+
+ATTR_CHARGE_MODE = "charge_mode"
+ATTR_SCHEDULES = "schedules"
+ATTR_TEMPERATURE = "temperature"
+ATTR_VIN = "vin"
+ATTR_WHEN = "when"
+
+REGEX_VIN = "(?i)^VF1[\\w]{14}$"
+
+SERVICE_AC_CANCEL_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_VIN): cv.matches_regex(REGEX_VIN),
```

Comment:
Consider replacing the vin parameter with device id selector for the car.

Example:
https://github.com/home-assistant/core/blob/88a08fdf57df12913071a26ace852588f2c716b8/homeassistant/components/system_bridge/services.yaml#L1-L11

https://github.com/home-assistant/core/blob/88a08fdf57df12913071a26ace852588f2c716b8/homeassistant/components/system_bridge/__init__.py#L115-L133